docs/content/concepts/assets/software-defined-assets.mdx

```diff
@@ -398,6 +398,43 @@ def downstream_asset(upstream):
 
 In this case, `ins={"upstream": AssetIn("upstream_asset")}` declares that the contents of the asset with the key `upstream_asset` will be provided to the function argument named `upstream`.
 
+### Attaching metadata
+
+Dagster supports attaching arbitrary metadata to assets. To attach arbitrary metadata, supply a `metadata` dictionary to the asset:
+
+```python file=/concepts/assets/asset_metadata.py startafter=start_simple_metadata endbefore=end_simple_metadata
+from dagster import asset
+
+@asset(
+    metadata={
+        "cereal_name": "Sugar Sprinkles"
+    },
+)
+```
+
+To specify data types for metadata values, use the <PyObject object="MetadataValue"/> class:
+
+```python file=/concepts/assets/asset_metadata.py startafter=start_typed_metadata endbefore=end_typed_metadata
+from dagster import asset, MetadataValue
+
+@asset(
+    metadata={
+        "cereal_name": "Sugar Sparkles",
+        "description": "Sparkly, sugary delight!",
+        "path": MetadataValue.path("/sugar_sparkles"),
+        "sales_dashboard_url": MetadataValue.url("http://mycoolsite.com/sugar_sparkle_sales")
+    },
+)
+```
+
+If a type is provided using <PyObject object="MetadataValue"/>, Dagster will use it to validate the metadata value.
+
+If a type isn't provided - such as `cereal_name` in the above example - Dagster will attempt validation using each supported <PyObject object="MetadataValue"/> type. If validation fails, the value will be typed as a string.
```

Comment:
I don't think this is the right page to describe metadata value normalization, since this information applies as well to every entity in dagster that takes metadata. Really we need a metadata concept page, but for now I would drop this material and just rely on link to `MetadataEntry` API doc per above comment.